pkg/objstore/inmem/inmem.go

```diff
@@ -86,6 +92,8 @@ func (b *Bucket) Get(_ context.Context, name string) (io.ReadCloser, error) {
 		return nil, errors.New("inmem: object name is empty")
 	}
 
+	b.mtx.RLock()
+	defer b.mtx.RUnlock()
```

Comment:
We could be a bit more specific with the unlocking here as there is no need to lock while reading the actual file.

Reply:
true, but we don't read - we just return the reader. ;p so it does not matter. It is not safe concurrent in case for deletions anyway

Reply:
yes exactly